src/vm/amd64/CrtHelpers.asm

```diff
@@ -13,516 +13,264 @@
+        mov     rdi, rcx                ; rdi is dst
+        mov     rcx, r8                 ; rcx is count
+        rep     stosb
+        mov     rdi, r11                ; restore rdi
+        mov     rax, r10
+        ret
+
+        align 16
+mset00: mov     rax, rcx                ; save dst address
+        movd    xmm0, rdx				
+        punpcklbw xmm0, xmm0            ; xmm0 is 16 bytes filler
+
+        cmp     r8, 128                
+        jbe     mset02  
+
+        ; count > 128 && count <= 512
+        mov     r9, r8
+        shr     r9, 7                   ; count/128
+        
+        align 16
+mset01: movdqu	[rcx], xmm0
+        movdqu	16[rcx], xmm0
+        movdqu	32[rcx], xmm0
+        movdqu	48[rcx], xmm0
+        movdqu	64[rcx], xmm0
+        movdqu	80[rcx], xmm0
+        movdqu	96[rcx], xmm0
+        movdqu	112[rcx], xmm0
+        add     rcx, 128
+        dec     r9
+        jnz     mset01    
+        and     r8, 7fh                 ; and r8 with 0111 1111
+        
+        ; the remainder is from 0 to 127
+        cmp     r8, 16                  
+        jnbe    mset02                  
+        
+        ; the remainder <= 16 
+        movdqu  -16[rcx + r8], xmm0
+        ret
+        
+        ; count > 16 && count <= 128 for mset02
+        align 16
+mset02: movdqu	[rcx], xmm0         
+        movdqu	-16[rcx + r8], xmm0    
+        cmp     r8, 32                 
+        jbe     mset03
+        
+        ; count > 32 && count <= 64
+        movdqu	16[rcx], xmm0
+        movdqu	-32[rcx + r8], xmm0
+        cmp     r8, 64
+        jbe     mset03
+        
+        ; count > 64 && count <= 128
+        movdqu	32[rcx], xmm0
+        movdqu	48[rcx], xmm0
+        movdqu	-48[rcx + r8], xmm0
+        movdqu	-64[rcx + r8], xmm0   
+mset03: ret
+ 
+        align 16
+mset04: mov     rax, rcx                ; save dst address
+        test    r8b, 24                 ; and r8b with 0001 1000
+        jz      mset05
+        
+        ; count >= 8 && count <= 16
+        mov     [rcx], rdx        
+        mov     -8[rcx + r8], rdx
+        ret
+
+        align 16
+mset05: test    r8b, 4                  ; and r8b with 0100
+        jz      mset06
+        
+        ; count >= 4 && count < 8
+        mov     [rcx], edx        
+        mov     -4[rcx + r8], edx
+        ret
+        
+        ; count >= 0 && count < 4
+        align 16
+mset06: test    r8b, 1                  ; and r8b with 0001
+        jz      mset07
+        mov     [rcx],dl
+mset07: test    r8b, 2                  ; and r8b with 0010
+        jz      mset08
+        mov     -2[rcx + r8], dx
+mset08: ret
 
 LEAF_END_MARKED JIT_MemSet, _TEXT
 
-;*******************************************************************************
-; This ensures that atomic updates of aligned fields will stay atomic. 
-;***
-;JIT_MemCpy - Copy source buffer to destination buffer
-;
-;Purpose:
-;JIT_MemCpy - Copy source buffer to destination buffer
```

Comment:
Can you please retain the header comments that describe the purpose of the method.
Please update the algorithm in the comments to reflect the new strategy.